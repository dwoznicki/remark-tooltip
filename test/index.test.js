const unified = require("unified");
const remarkParse = require("remark-parse");
const remarkTooltip = require("../index");
const remark2rehype = require("remark-rehype");
const rehypeStringify = require("rehype-stringify");

const stringify = doc => {
    return unified()
        .use(remarkParse)
        .use(remarkTooltip)
        .use(remark2rehype)
        .use(rehypeStringify)
        .processSync(doc).contents;
};

describe("HTML tooltip", () => {
    it("should parse markdown", () => {
        const doc = "[base text]^tooltip text^";
        expect(stringify(doc)).toMatchSnapshot();
    });

    it("should parse markdown within the tooltip", () => {
        const doc = "[base text]^*tooltip* [content](/content)^";
        expect(stringify(doc)).toMatchSnapshot();
    });

    it("should ignore escaped brackets (\\]) and carets (\\^)", () => {
        const doc = "[base \\[\\] text]^tooltip \\^ text^";
        expect(stringify(doc)).toMatchSnapshot();
    });

    it("should render properly inline", () => {
        const doc = "He's a *handsome* [gecko]^Anaximander^.";
        expect(stringify(doc)).toMatchSnapshot();
    });
});

