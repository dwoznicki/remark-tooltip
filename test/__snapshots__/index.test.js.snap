// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML tooltip should ignore escaped brackets (\\]) and carets (\\^) 1`] = `"<p><span class=\\"tooltip\\">base \\\\[\\\\] text<div class=\\"tooltip-popup\\">tooltip \\\\^ text</div></span></p>"`;

exports[`HTML tooltip should parse markdown 1`] = `"<p><span class=\\"tooltip\\">base text<div class=\\"tooltip-popup\\">tooltip text</div></span></p>"`;

exports[`HTML tooltip should parse markdown within the tooltip 1`] = `"<p><span class=\\"tooltip\\">base text<div class=\\"tooltip-popup\\"><em>tooltip</em> <a href=\\"/content\\">content</a></div></span></p>"`;

exports[`HTML tooltip should render properly inline 1`] = `"<p>He's a <em>handsome</em> <span class=\\"tooltip\\">gecko<div class=\\"tooltip-popup\\">Anaximander</div></span>.</p>"`;
